compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -101,6 +101,17 @@ impl SirFuncCx<'tcx> {
                     );
                 }
 
+                let inst_kind = instance.ty(tcx, ty::ParamEnv::reveal_all()).kind();
+                if !matches!(inst_kind, ty::FnDef(..)) {
+                    // We can't accept closures, as these have a hidden argument prepended for the
```

Comment:
I *think* this comment should be something like "We can't accept anything that's not a normal function definition (e.g. we can't accept closures because of their hidden upvars argument)".

Reply:
That's fine by me.